src/flood_wave_core/flood_wave_handler.py

```diff
@@ -63,19 +64,82 @@ def find_dates_for_next_gauge(
     def convert_datetime_to_str(
             actual_date: datetime,
             gauge_pair: dict,
-            next_gauge_dates: pd.DataFrame
+            next_gauge_dates: pd.DataFrame,
+            slopes
     ) -> None:
         """
         Converts the date(s) to our desired format string. Then the list of converted strings is stored in a dictionary
 
         :param datetime actual_date: The date to be converted
         :param dict gauge_pair: A dictionary to store the converted list of strings
         :param pd.DataFrame next_gauge_dates: A DataFrame containing the found dates to be converted
+        :param slopes: slope or slopes between the two vertices
         """
 
         if not next_gauge_dates.empty:
             found_next_dates_str = next_gauge_dates['Date'].dt.strftime('%Y-%m-%d').tolist()
-            gauge_pair[actual_date.strftime('%Y-%m-%d')] = found_next_dates_str
+            gauge_pair[actual_date.strftime('%Y-%m-%d')] = (found_next_dates_str, slopes)
+
+    @staticmethod
+    def preprocess_for_get_slopes(current_gauge: str, next_gauge: str, river_kms: pd.DataFrame, folder_name: str):
+        """
+        This is a helper function for FloodWaveHandler.get_slopes()
+        :param str current_gauge: gauge number of current gauge as a string
+        :param str next_gauge: gauge number of current gauge as a string
+        :param pd.DataFrame river_kms: dataframe of river kilometers
+        :param str folder_name: name of the data folder as a string
+        :return tuple: current_vertices, next_vertices, current_null, next_null, distance
+        """
+        null_points = JsonHelper.read(os.path.join(PROJECT_PATH, 'data', 'nullpontok_fontos.json'))
+        current_null = null_points[current_gauge]
+        next_null = null_points[next_gauge]
+
+        current_vertices = JsonHelper.read(os.path.join(PROJECT_PATH, folder_name,
+                                                        'find_vertices', f'{current_gauge}.json'))
+        next_vertices = JsonHelper.read(os.path.join(PROJECT_PATH, folder_name,
+                                                     'find_vertices', f'{next_gauge}.json'))
+
+        current_river_km = river_kms[float(current_gauge)]
+        next_river_km = river_kms[float(next_gauge)]
+
+        distance = float(next_river_km - current_river_km)
+
+        return current_vertices, next_vertices, current_null, next_null, distance
```

Comment:
Having 5 return values is a very bad practice. Moreover as I see in `FloodWaveDetector.find_edges`, these return values are used only in `FloodWaveHandler.get_slopes`, which gives the idea to create a class for slope calculation, where these return values can be member variables. I would instantiate a new `SlopeCalculator` class at every iteration of the `for` loop. This will lead to clearer calls in `find_edges` keeping its readability.